system/core/tags.php

```diff
@@ -0,0 +1,184 @@
+<?php
+	// TODO:
+	// - add styling for default theme
+	// - implement error notification if new tag exists
+	// - implement tag management page: edit, translate, remove or merge tags
+	// - implement tagcontainer pagetype
+	
+	function tagList($pageListNode, $lang) {
+		global $hyphaXml;
+		
+		// Build list of active tags for given page
+		foreach($pageListNode->getElementsByTagName('tag') as $tag) $activeTags[] = $tag->getAttribute('id');
+		
+		// Divide list of all tags stored in hypha.xml in a selected tag list and a deselected tag list
+		foreach($hyphaXml->getElementsByTagName('tag') as $tag) {
+			foreach($tag->getElementsByTagName('language') as $tagLang) {
+				if ($tagLang->getAttribute('id')==$lang) {
+					if (!empty($activeTags) && in_array($tag->getAttribute('id'), $activeTags)) $selectedTags[$tag->getAttribute('id')] = array("label" => $tagLang->getAttribute('label'), "description" => $tagLang->nodeValue);
```

Comment:
This seems like an overly long line to me, I'd add some line break(s).

Also, I think that if you assign an empty array to `$activeTags` at the start of the function, you won't have to do the `empty` check here.

Reply:
Fixed by Laurens.